tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -835,6 +836,25 @@ LogicalResult ConvertTFLSignOp::matchAndRewrite(
   return success();
 }
 
+LogicalResult ConvertTFLCastOp::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_cast_op = cast<TFL::CastOp>(op);
+
+  RankedTensorType input_type =
+      tfl_cast_op.getInput().getType().dyn_cast<RankedTensorType>();
+  RankedTensorType output_type =
+      tfl_cast_op.getOutput().getType().dyn_cast<RankedTensorType>();
+  // Not a ranked tensor input or output
+  if (!input_type || !output_type) return failure();
```

Comment:
Instead of `return failure()` please include `notifyMatchFailure` debug statements. It makes failure cases easier to debug.

Reply:
done